R/read_csv.R

```diff
@@ -370,9 +373,112 @@ read_sample_csv <- function(files,
   read_cmdstan_csv(files, variables, sampler_diagnostics)
 }
 
+#' @rdname read_cmdstan_csv
+#' @export
+#' @param csv_contents The list returned by `read_cmdstan_csv()`.
+#' @param check_diagnostics Should diagnostic checks be performed? The default
+#'   is `TRUE` but set to `FALSE` to avoid checking diagnostics like
+#'   divergences.
+#' @examples
+#' \dontrun{
+#' # create a CmdStanMCMC object
+#' x <- read_cmdstan_csv(fit$output_files())
+#' fit <- as_cmdstan_mcmc(x)
+#' }
+as_cmdstan_mcmc <- function(csv_contents, check_diagnostics = TRUE) {
+  CmdStanMCMC2$new(csv_contents, check_diagnostics)
+}
+
+#' @rdname read_cmdstan_csv
+#' @export
+as_cmdstan_mle <- function(csv_contents) {
+  CmdStanMLE2$new(csv_contents)
+}
+
+#' @rdname read_cmdstan_csv
+#' @export
+as_cmdstan_vb <- function(csv_contents) {
+  CmdStanVB2$new(csv_contents)
+}
+
+
 
 # internal ----------------------------------------------------------------
 
+#' Create a CmdStanMCMC/MLE/VB-ish objects from `read_cmdstan_csv()` output
+#' instead of from a CmdStanRun object
+#'
+#' The resulting object has fewer methods than a CmdStanMCMC/MLE/VB object
+#' because it doesn't have access to the CSV files directly.
+#'
+#' @noRd
+#'
+CmdStanMCMC2 <- R6::R6Class(
+  classname = "CmdStanMCMC2",
+  inherit = CmdStanMCMC,
```

Comment:
Don't actually create CmdStanMCMC objects but CmdStanMCMC2 objects that inherit from CmdStanMCMC. Same approach with MLE and VB below. This avoids having to create the runset object because we can override the `initialize` method.  

Suggestions for better class name than `CmdStanMCMC2`?

Reply:
What aobut CmdStanMCMCCSV, CmdStanMCMC_CSV? I am bad for names.